core/src/main/java/com/linecorp/armeria/client/cookie/CookieJar.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (c) 2005, 2012, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.linecorp.armeria.client.cookie;
+
+import java.net.URI;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.Cookie;
+import com.linecorp.armeria.common.Cookies;
+
+/**
+ * A {@link Cookie} container for a client.
+ */
+public interface CookieJar {
+
+    /**
+     * Determines whether a host matches a domain, as specified by RFC 6265
+     * <a href="https://tools.ietf.org/html/rfc6265#section-5.1.3">Domain Matching</a>.
+     */
+    static boolean domainMatches(@Nullable String domain, @Nullable String host) {
+        // Forked from OpenJDK
```

Comment:
OpenJDK could be redistributed only under GNU General Public License version 2.
However, Armeria distributes under Apache License version 2.0
You have to just refer to the code or fork from other open source projects that are distributed under Apache License version 2.0 such as OkHttp. 
https://github.com/square/okhttp/blob/fd509b937c4a5438fe02ddf06ade5ad56736af5d/okhttp/src/main/java/okhttp3/Cookie.java#L182

Reply:
Oh so we don't need to include the license and the comments? Got it.

Reply:
We can not modify JDK source code and use it in the Armeria. 😢

Reply:
Oh I see. So can I rewrite it in a way that functions the same as JDK? I mean not syntactically but functionally similar?

Reply:
Actually I think it's better if I look at okhttp as a reference point

Reply:
👍 That would be a better way to avoid license issues.